src/Calculator/Resources/en-US/Resources.resw

```diff
@@ -4106,7 +4106,167 @@
+  <data name="FactorialCannotPerformOnLargeNumber" xml:space="preserve">
+    <value>The factorial function has an argument that is too large to graph.</value>
+    <comment>Error that occurs during graping when a factorial has a large n</comment>
+  </data>
+  <data name="ModuloCannotPerformOnFloat" xml:space="preserve">
+    <value>Modulo can only be used with whole numbers.</value>
+    <comment>Error that occurs during graping when modulo is used with a float.</comment>
+  </data>
+  <data name="EquationTooComplexToSolve" xml:space="preserve">
+    <value>The equation is too complex to graph.</value>
+    <comment>Error that occurs during graping when the equation is too complex to solve.</comment>
+  </data>
+  <data name="EquationHasNoSolution" xml:space="preserve">
+    <value>The equation has no solution.</value>
+    <comment>Error that occurs during graping when the equation has no solution.</comment>
+  </data>
+  <data name="DivideByZero" xml:space="preserve">
+    <value>Cannot divide by zero.</value>
+    <comment>Error that occurs during graping when a divison by zero occurs.</comment>
+  </data>
+  <data name="MutuallyExclusiveConditions" xml:space="preserve">
+    <value>The equation contains logical conditions that are mutually exclusive.</value>
+    <comment>Error that occurs during graping when mutually exclusive conditions are used.</comment>
+  </data>
+    <data name="OutOfDomain" xml:space="preserve">
+    <value>Equation is out of domain.</value>
+    <comment>Error that occurs during graping when the equation is out of domain.</comment>
+  </data>
+    <data name="GE_NotSupported" xml:space="preserve">
+    <value>Graphing this equation is not supported.</value>
+    <comment>Error that occurs during graping when the equation is not supported.</comment>
+  </data>
+  <data name="ParenthesisMismatch" xml:space="preserve">
+    <value>The equation is missing an opening paranthesis.</value>
+    <comment>Error that occurs during graping when the equation is missing a (</comment>
+  </data>
+  <data name="UnmatchedParenthesis" xml:space="preserve">
+    <value>The equation is missing a closing paranthesis.</value>
+    <comment>Error that occurs during graping when the equation is missing a )</comment>
+  </data>
+  <data name="TooManyDecimalPoints" xml:space="preserve">
+    <value>Too many decimal points in a number.</value>
+    <comment>Error that occurs during graping when a number has too many decimals. Ex: 1.2.3</comment>
+  </data>
+  <data name="DecimalPointWithoutDigits" xml:space="preserve">
+    <value>Decimal point is missing digits.</value>
+    <comment></comment>
+  </data>
+  <data name="UnexpectedEndOfExpression" xml:space="preserve">
+    <value>Unexpected end of expression.</value>
+    <comment>Error that occurs during graping when the expression ends unexpectedly. Ex: 3-4*</comment>
+  </data>
+  <data name="UnexpectedToken" xml:space="preserve">
+    <value>Unexpected token in the expression.</value>
```

Comment:
I don't think most users would know how to fix the expression because I don't think they will know what a token is. Can we give an example of what might be the problem or give a different error message?

Reply:
How about "Unexpected characters in the expression"